tap_s3_csv/sync.py

```diff
@@ -159,8 +159,11 @@ def sync_csv_file(config, file_handle, s3_path, table_spec, stream):
     # memory consumption but that's acceptable as well.
     csv.field_size_limit(sys.maxsize)
 
-    iterator = csv_helper.get_row_iterator(
-        file_handle, table_spec, stream["schema"]["properties"].keys(), True)
+    if "properties" in stream["schema"]:
+        iterator = csv_helper.get_row_iterator(
+            file_handle, table_spec, stream["schema"]["properties"].keys(), True)
+    else:
+        iterator = csv_helper.get_row_iterator(file_handle, table_spec, None, True)
```

Comment:
When all five CSV files used in sampling are without any records, the catalog will be generated without a properties field so emitting all the records when properties are not found.